### PR TITLE
chore: Add issue template for Argo CD onboarding

### PR DESCRIPTION
**Pull request by tumido**
Part of: #39 

Adds an issue template checklist/questionnaire for Argo CD onboarding

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/onboarding_argocd.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+name: Onboard to Continuous Delivery
```
[the remark this answers is not in the document]
Comment (tumido) [reply .github/ISSUE_TEMPLATE/onboarding_argocd.md:2]: I think CD means Continous Delivery, your're not deploying it all the time, your delivering it all the time. :smile: 

https://www.redhat.com/en/topics/devops/what-is-ci-cd#:~:text=The%20%22CD%22%20in%20CI%2F,how%20much%20automation%20is%20happening.

File: .github/ISSUE_TEMPLATE/onboarding_argocd.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+#### Creating a new project
+
+1. **Team name**:
+   Please tell us your team name, which corresponds to [an already available user `Group`](https://github.com/operate-first/apps/tree/master/cluster-scope/base/groups).
+
+2. **Target cluster**:
+   Choose an environment from [this list](https://github.com/operate-first/apps/tree/master/cluster-scope/overlays)
+
+3. **Managed namespaces**:
```
[the remark this answers is not in the document]
Comment (tumido) [reply .github/ISSUE_TEMPLATE/onboarding_argocd.md:26]: I think this is a prerequisite. I'll refer to the other template then. :+1:

[the remark this answers is not in the document]
Comment (tumido) [reply .github/ISSUE_TEMPLATE/onboarding_argocd.md:26]: Or, wait a minute.. this is the Create project section. Here we'd like to ask them to specify a namespace prefix, that we would wildcard match against. The namespaces don't need to exist yet.

File: .github/ISSUE_TEMPLATE/onboarding_argocd.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (anishasthana) [reply .github/ISSUE_TEMPLATE/onboarding_argocd.md:2]: Ugh, I hate how it's not at all consistent. I remember attending a Red Hat presentation on TektonCD and they said it depends on context, but generally think of delivery as things like image builds and all that, while deployment is for actual objects in k8s. (I'm honestly fine either way)

File: .github/ISSUE_TEMPLATE/onboarding_argocd.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (anishasthana) [reply .github/ISSUE_TEMPLATE/onboarding_argocd.md:26]: Ah good point, didn't realise it's "creating a new project"

**Review by anishasthana — APPROVED** [review]
Looks good to me.

File: .github/ISSUE_TEMPLATE/onboarding_argocd.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+labels: onboarding
+assignees: ""
+---
+
+### Requested actions
+
+Select all that apply:
+
+- [ ] Create a new ArgoCD project for existing team
+- [ ] Deploy an application
```
Comment (goern) [inline .github/ISSUE_TEMPLATE/onboarding_argocd.md:14]: ist "Create" required if I do "Deploy" ?

**Review by goern — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/onboarding_argocd.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tumido) [reply .github/ISSUE_TEMPLATE/onboarding_argocd.md:2]: I think you're right though.. I can agree with this description:
https://www.atlassian.com/continuous-delivery/principles/continuous-integration-vs-delivery-vs-deployment 
I'll change it. :wink:

File: .github/ISSUE_TEMPLATE/onboarding_argocd.md (hunk 3 above)
Comment (tumido) [reply .github/ISSUE_TEMPLATE/onboarding_argocd.md:14]: Good catch.. we should change the wording here. :+1:

Comment (tumido) [reply .github/ISSUE_TEMPLATE/onboarding_argocd.md]: What about this: `Enlist an application to be deployed automatically via ArgoCD`

File: .github/ISSUE_TEMPLATE/onboarding_argocd.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+#### Creating a new project
+
+1. **Team name**:
+   Please tell us your team name, which corresponds to [an already available user `Group`](https://github.com/operate-first/apps/tree/master/cluster-scope/base/groups).
+
+2. **Target cluster**:
+   Choose an environment from [this list](https://github.com/operate-first/apps/tree/master/cluster-scope/overlays)
+
+3. **Managed namespaces**:
+   The target namespaces that should be managed by ArgoCD. Preferably using a single (team name) prefix.
```
Comment (HumairAK) [inline .github/ISSUE_TEMPLATE/onboarding_argocd.md:27]: Can we link to [here](https://github.com/operate-first/argocd-apps/blob/main/docs/onboard_team_to_argocd.md#additional-notes) to for further explanation of why a prefix is preferred.

**Review by HumairAK — APPROVED** [review]

**Pull request merged**